README.md

```diff
@@ -1,53 +1,294 @@
+In both cases you don't want to map the fpm port to you host. 
+
+```console
+$ docker run -d nextcloud-fpm
+```
 
+As the fastCGI-Process is not capable of serving static files (style sheets, images, ...) the webserver needs access to these files. This can be achieved with the `volumes-from` option. You can find more information in the docker-compose section.
 
-By default this container uses SQLite for data storage, but the Nextcloud setup wizard (appears on first run) allows connecting to an existing MySQL/MariaDB or PostgreSQL database. You can also link a database container, e.g. `--link my-mysql:mysql`, and then use `mysql` as the database host on setup.
+## Using an external database
+By default this container uses SQLite for data storage, but the Nextcloud setup wizard (appears on first run) allows connecting to an existing MySQL/MariaDB or PostgreSQL database. You can also link a database container, e.g. `--link my-mysql:mysql`, and then use `mysql` as the database host on setup. More info is in the docker-compose section.
 
 ## Persistent data
+The nextcloud installation and all data beyond what lives in the database (file uploads, etc) is stored in the [unnamed docker volume](https://docs.docker.com/engine/tutorials/dockervolumes/#adding-a-data-volume) volume `/var/www/html`. The docker daemon will store that data within the docker directory `/var/lib/docker/volumes/...`. That means your data is saved even if the container crashes, is stopped, updated or deleted.
 
-All data beyond that which lives in the database (file uploads, etc) is stored within the default volume `/var/www/html`. With this volume, Nextcloud will only be updated when the file `version.php` is not present.
+When nextcloud is updated the source files and system apps, that are shipped with the nextcloud installation are overwritten. Then, an upgrade routine that upgrades the database and handles your data, is started.
 
-- `-v /<mydatalocation>:/var/www/html`
+To backup your nextcloud installation, you have to take care of five parts: data, config, apps, theming and database. 
+The data and config are stored in respective subfolders inside `/var/www/html/`. The apps are split into system apps (wich are shipped with nextcloud and you don't need to take care of) and a `custom_apps` folder. If you want to use a custom theme, you can use the `theming` folder.
 
-For fine grained data persistence, you can use 3 volumes, as shown below.
+Overview of the folders:
 
-- `-v /<mydatalocation>/apps:/var/www/html/apps` installed / modified apps
-- `-v /<mydatalocation>/config:/var/www/html/config` local configuration
-- `-v /<mydatalocation>/data:/var/www/html/data` the actual data of your Nextcloud
+- `/var/www/html/custom_apps` installed / modified apps
+- `/var/www/html/config` local configuration
+- `/var/www/html/data` the actual data of your Nextcloud
+- `/var/www/html/theming` theming/branding
```

Comment:
Yeah need to rework that section too (mounting the whole folder is not necessary because of the examples folder). Also #72